### PR TITLE
[commands] update clean_content to work when standalone

### PR DESCRIPTION
**Pull request by z03h**
## Summary

Changes clean_content converter to not use `ctx.message.raw_mentions`/`role_mentions` so it gives more expected results when used standalone outside of command conversion. Also fixes bug where specifying `use_nicknames=False` could resolve users outside of the current guild.

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by Rapptz** [comment]
Was it necessary to rewrite the entire function for this supposed bug fix? `clean_content` used those attributes by design because the `mentions` field requires the users to actually be in the mention regardless of `allowed_mentions`. The point of the function is to clean *all* mentions, even potential ones that are ignored from `allowed_mentions`. Likewise, `role_mentions` field requires role cache and has the same gotchas as the one I mentioned for users. I see no real reason to change the attributes.

**Comment by z03h** [comment]
The rewrite was to make it searches arg for mentions instead of only replacing mentions that are in ctx.message.content.
Intention was to add ability to clean mentions if you use it outside of type hints like `await clean_content().convert(ctx, random_string)`. Unless that's an unintended use for converters?

File: discord/ext/commands/converter.py [hunk 1]
```diff
@@ -818,67 +818,58 @@ class clean_content(Converter[str]):
...
+                      '@&': resolve_role,
+                      }
+
+        def repl(match):
+            type = match[1]
+            id = int(match[2])
+            transformed = transforms[type](id)
+            return transformed or match[0]
+
+        pattern = re.compile(r'<(@[!&]?|#)(\d{15,20})>')
```
Comment (Rapptz) [inline discord/ext/commands/converter.py:871]:
```suggestion
        pattern = re.compile(r'<(@[!&]?|#)([0-9]{15,20})>')
```

If this compilation isn't stored then it's pointless, the `re` module already caches recently used regex.

File: discord/ext/commands/converter.py [hunk 2]
```diff
@@ -818,67 +818,58 @@ class clean_content(Converter[str]):
...
+                return f'#{c.name}' if c else '#deleted-channel'
+        else:
+            def resolve_channel(id):
+                return
+
+        transforms = {'@': resolve_member,
+                      '@!': resolve_member,
+                      '#': resolve_channel,
+                      '@&': resolve_role,
+                      }
```
Comment (Rapptz) [inline discord/ext/commands/converter.py:863]:
```suggestion
        transforms = {
            '@': resolve_member,
            '@!': resolve_member,
            '#': resolve_channel,
            '@&': resolve_role,
        }
```

File: discord/ext/commands/converter.py [hunk 3]
```diff
@@ -818,67 +818,58 @@ class clean_content(Converter[str]):
         .. versionadded:: 1.7
     """
 
-    def __init__(self, *, fix_channel_mentions: bool = False, use_nicknames: bool = True, escape_markdown: bool = False, remove_markdown: bool = False) -> None:
+    def __init__(self,
+                 *,
+                 fix_channel_mentions: bool = False,
+                 use_nicknames: bool = True,
+                 escape_markdown: bool = False,
+                 remove_markdown: bool = False) -> None:
```
Comment (Rapptz) [inline discord/ext/commands/converter.py:826]:
```suggestion
    def __init__(
        self,
        *,
        fix_channel_mentions: bool = False,
        use_nicknames: bool = True,
        escape_markdown: bool = False,
        remove_markdown: bool = False,
    ) -> None:
```

File: discord/ext/commands/converter.py [hunk 4]
```diff
@@ -818,67 +818,58 @@ class clean_content(Converter[str]):
...
-            (f'<@{member_id}>', resolve_member(member_id))
-            for member_id in message.raw_mentions
-        )
-
-        transformations.update(
-            (f'<@!{member_id}>', resolve_member(member_id))
-            for member_id in message.raw_mentions
-        )
-        # fmt: on
+    async def convert(self, ctx, arg):
```
Comment (Rapptz) [inline discord/ext/commands/converter.py:832]:
```suggestion
    async def convert(self, ctx: Context, argument: str) -> str:
```

Changing this signature made no sense.

**Review by Rapptz — CHANGES_REQUESTED** [review]
Overall this is missing type hints. This also doesn't match the style of the rest of the library.

**Comment by z03h** [comment]
🙃 I'm dented

File: discord/ext/commands/converter.py [hunk 5]
```diff
@@ -818,67 +818,60 @@ class clean_content(Converter[str]):
...
-            (f'<@{member_id}>', resolve_member(member_id))
-            for member_id in message.raw_mentions
-        )
-
-        transformations.update(
-            (f'<@!{member_id}>', resolve_member(member_id))
-            for member_id in message.raw_mentions
-        )
-        # fmt: on
+    async def convert(self, ctx: Context, arg: str) -> str:
```
Comment (Rapptz) [inline discord/ext/commands/converter.py:834]:
```suggestion
    async def convert(self, ctx: Context, argument: str) -> str:
```

Parameter names are part of the API contract, you can't change them.

**Review by Rapptz — CHANGES_REQUESTED** [review]
Type hints are still missing but it's fine it can be dealt with afterwards if you're not familiar with it.

**Comment by z03h** [comment]
ah I missed that, and type hinting all the `resolve_x`?

**Comment by Rapptz** [comment]
If you're not familiar with it it's fine.

**Comment by z03h** [comment]
I think that's it, not sure if I missed something

**Review by Rapptz — APPROVED** [review]

**Pull request merged**